perceval/backends/core/confluence.py

```diff
@@ -86,11 +86,22 @@ def fetch(self, category=CATEGORY_HISTORICAL_CONTENT, from_date=DEFAULT_DATETIME
 
         from_date = datetime_to_utc(from_date)
 
-        kwargs = {'from_date': from_date}
+        kwargs = {
+            'from_date': from_date
+        }
+
         items = super().fetch(category, **kwargs)
 
         return items
 
+    @staticmethod
+    def ancestors(item, content):
+        ancestors = []
+        if ('ancestors' in content):
```

Comment:
tests coverage went done due to this code, a test is needed when no ancestors are returned.
I would put something:
```
if 'ancestors' not in content:
 return []

...
```

Reply:
I really don't see the difference between the 2 syntaxes, since the guard is exactly the same.

I believe this code is less readable, but may be personal preference:

```
    def __fetch_ancestors(self, item, content):
        if ('ancestors' not in content):
            return []
        else:
            ancestors = []
            for ancestor in content['ancestors']:
                ancestors.append(ancestor['_links']['webui'])
            return ancestors
```

If this is not what you meant, can you please share your own version? Maybe I'm not understanding your point here.

Reply:
yes, sure! the idea in Python is to reduce the number of indentation as much as possible (https://stackoverflow.com/questions/10959683/preferred-maximum-indentation-in-python

In the code you provide above, the else statement is not needed. If the if-condition is hit, the program will return after executing the nested statement. Conversely, if the if-condition is not true, the program will ignore that part and go directly to the next statement.

```
def __fetch_ancestors(self, item, content):
        ancestors = []
        if 'ancestors' not in content: # parenthesis are not needed
            return ancestors
       
       # the code below should go to the client, to avoid mixing logic of different level of abstraction (in a nutshell, the backend should not know the low level-details of the data returned by the API)
       for ancestor in content['ancestors']:
             ancestors.append(ancestor['_links']['webui'])
       return ancestors
```